admin_guide/out_of_resource_handling.adoc

```diff
@@ -0,0 +1,270 @@
+
+* `eviction-pressure-transition-period` is the duration for which the node has
+to wait before transitioning out of an eviction pressure condition.
+
+The node would ensure that it has not observed an eviction threshold being met
+for the specified pressure condition for the period specified before toggling the
+condition back to `false`.
+
+=== Eviction of Pods
+
+If an eviction threshold has been met and the grace period has passed,
+the node will initiate the process of evicting pods until it has observed 
+the signal has gone below its defined threshold.
+
+The node ranks pods for eviction 1) by their quality of service,
+2) and among those with the same quality of service by the consumption of the
+starved compute resource relative to the pods scheduling request.
+
+* `BestEffort` pods that consume the most of the starved resource are failed
+first.
+* `Burstable` pods that consume the greatest amount of the starved resource
+relative to their request for that resource are killed first.  If no pod
+has exceeded its request, the strategy targets the largest consumer of the
+starved resource.
+* `Guaranteed` pods that consume the greatest amount of the starved resource
+relative to their request are killed first.  If no pod has exceeded its request,
+the strategy targets the largest consumer of the starved resource.
+
+A `Guaranteed` pod is guaranteed to never be evicted because of another pod's
+resource consumption.  If a system daemon (i.e. node, `docker`, `journald`, etc.)
+is consuming more resources than were reserved via `system-reserved` or `kube-reserved` allocations,
+and the node only has `Guaranteed` pod(s) remaining, then the node must choose to evict a
+`Guaranteed` pod in order to preserve node stability, and to limit the impact
+of the unexpected consumption to other `Guaranteed` pod(s).
+
+=== Scheduler
+
+The node will report a condition when a compute resource is under pressure.  The
+scheduler views that condition as a signal to dissuade placing additional
+pods on the node.
+
+.Node Conditions and Scheduler Behavior
+[cols="3a,8a",options="header"]
+|===
+
+|Node Condition |Scheduler Behavior
+
+|`*MemoryPressure*`
+| `BestEffort` pods are not scheduled to the node.
+|===
+
+== Node OOM Behavior
+
+If the node experiences a system OOM (out of memory) event prior to the node is able to reclaim memory,
+the node depends on the [oom_killer](https://lwn.net/Articles/391222/) to respond.
+
+The node sets a `oom_score_adj` value for each container based on the quality of service for the pod.
+
+.Quality of Service OOM Scores
+[cols="3a,8a",options="header"]
+|===
+
+| Quality of Service |oom_score_adj
+
+| `Guaranteed`
+| -998
+
+| `BestEffort`
+| 1000
+
+| `Burstable`
+| min(max(2, 1000 - (1000 * memoryRequestBytes) / machineMemoryCapacityBytes), 999)
+|===
+
+If the node is unable to reclaim memory prior to a node experiencing system OOM, the `oom_killer` will calculate
```

Comment:
If the node is unable to reclaim memory prior to experiencing a system OOM event